app/dependencies.py

```diff
@@ -22,13 +23,30 @@ async def get_users(db: AsyncSession = Depends(get_db)):
 from app.database import SessionLocal
 from app.models import models_core
 from app.schemas import schemas_auth
+from app.utils.redis import connect
 from app.utils.tools import is_user_member_of_an_allowed_group
 from app.utils.types.groups_type import GroupType
 from app.utils.types.scopes_type import ScopeType
 
 # We could maybe use hyperion.security
 hyperion_access_logger = logging.getLogger("hyperion.access")
 
+redis_client = None  # Create a global variable for the redis client, so that it can be instancied in the startup and shutdown events
+# Is None if the redis client is not instancied, is False if the redis client is instancied but not connected, is a redis.Redis object if the redis client is connected
+
+
+def get_redis_client(
+    settings=None,
+) -> redis.Redis | None | bool:  # settings can be None if the redis client is already instancied, so that we don't need to pass the settings to the function
+    """
+    Dependency that return the redis client
+    """
```

Comment:
Putting the comment in the docstring would make it easier to see for people using the dependency

```suggestion
) -> redis.Redis | None | bool:
    """
    Dependency that return the redis client
    
    Settings can be None if the redis client is already instanced, so that we don't need to pass the settings to the function.

    Is None if the redis client is not instancied, is False if the redis client is instancied but not connected, is a redis.Redis object if the redis client is connected
    """
```